### PR TITLE
Added default values for error.error and error.message

### PR DESCRIPTION
**Pull request by TatriX**
When tivoka used with http://golang.org/pkg/net/rpc/jsonrpc/ there is no "error" or "message" in server response, so we get and undefined index exception. Fixed it by providing some default values.

### CODE REVIEW

File: lib/Tivoka/Client/Request.php [hunk 1]
```diff
@@ -140,8 +140,8 @@ public function setHeaders($raw_headers) {
     public function interpretResponse($json_struct) {
         //server error?
         if(($error = self::interpretError($this->spec, $json_struct, $this->id)) !== FALSE) {
-            $this->error        = $error['error']['code'];
-            $this->errorMessage = $error['error']['message'];
+            $this->error        = (isset($error['error']['error'])) ? $error['error']['code'] : 'error';
```
Comment (BugBuster1701) [inline lib/Tivoka/Client/Request.php]: I think, you meen "code" here:

    $this->error = (isset($error['error']['code'])) ? $error['error']['code'] : 'error';

**Comment by rafalwrzeszcz** [comment]
I'm against this - http://www.jsonrpc.org/specification#error_object states clearly that response has to have these properties. If Go extension doesn't return it, it's Go extension error.

**Comment by TatriX** [comment]
Well its a go's standard library implementation. It uses 1.0 spec, and I cannot find anything about error object there: http://json-rpc.org/wiki/specification

File: lib/Tivoka/Client/Request.php (hunk 1 above)
Comment (TatriX) [inline lib/Tivoka/Client/Request.php]: Yeah, of course. My bad.

**Comment by fiddur** [comment]
Hmm, this PR is included in the other PR as well.  As I commented there, `null` would be a better default for representing "no value", if one is really wanted.

But, since 1.0 does not specify the content of the error member, perhaps it is better to not try to extract code, message and data, but just give access to the error as it is.

**Pull request closed without merging**